tensorboard/plugins/text/text_plugin.py

```diff
@@ -166,8 +166,7 @@ def text_array_to_html(text_arr):
   """
   if not text_arr.shape:
     # It is a scalar. No need to put it in a table, just apply markdown
-    return plugin_util.markdown_to_safe_html(
-        text_arr.astype(np.dtype(str)).tostring())
+    return plugin_util.markdown_to_safe_html(np.asscalar(text_arr))
```

Comment:
Great - it looks like `np.asscalar(text_arr)` returns a `str` (raw bytes representation) in python2 and `str` (unicode) in python3. Either way, `markdown_to_safe_html` now gracefully deals with the value.

And we avoid encoding in a way that assumes little-endian UTF-32 (woah) as noted in #647.

Reply:
Yeah, the point was just to make the treatment for scalars match what's being done for all the non-scalar cases where they're converted into HTML element-wise (and then concatenated).  Numpy will natively represent the strings as you described, str == bytes in py2 and str == unicode in py3, so I figured I wouldn't change that.

Using `np.asscalar(a)` is literally just syntatic sugar for doing `a.item()` but it's slightly more explicit, so I went with that.